pkg/apis/core/v1/helper/helpers.go

```diff
@@ -47,13 +47,18 @@ func IsExtendedResourceName(name v1.ResourceName) bool {
 	return true
 }
 
+// IsDefaultnamespaceContainingResource returns true if the resource name is in the
+// *kubernetes.io/ namespace.
+func IsDefaultNamespaceContainingResource(name v1.ResourceName) bool {
```

Comment:
The name of this function is confusing and the fact that we have used "Namespace" which is a well known concept in Kubernetes causes even more confusion. I would call it something like "IsNativeResource".

Reply:
I based it on the function below: `IsDefaultNamespaceResource90`. Also, the prefix variable is called `ResourceDefaultNamespacePrefix`.

How would you name these two functions? `IsNativeResource()` seems to make more sense for the function below.

Since the newly added function is very straightforward and seem to not have uses outside of this feel, I don't mind inlining it either.

Reply:
I know that this whole "Namespace" thing in the function names existed.
You are right that we should probably use "IsNativeResource" for the function below. This one can be called "IsNativeResourceWithPrefix" or "IsPrefixedNativeResource" or something along these.

Reply:
Updated. PTAL